### DISCUSSION TITLE
how to connect two exsting records in many to many realtionship?

### DISCUSSION THREAD

**Discussion by kraighamady** [prisma/prisma#20107] [Q&A]
I have a schema with an explicit many to many relationship.  In my use case I create a person in one step and later on I create a team. In a later interaction, I want to create a link (connection) between the two.  

I have seen lots of examples in the docs for making the connection while creating one or the other (player or team), but no examples of connecting two existing records.

Here are my models:
```Prisma
model Person {
  id                      Int                      @id @default(autoincrement())
  teamMember   TeamMember[]
}

model Team {
  id                     Int                      @id @default(autoincrement())
  teamMember  TeamMember[]
}

model TeamMember {
  id               Int                   @id @default(autoincrement())
  teamId       Int                   @map("team_id")
  team          Team               @relation(fields: [teamId], references: [id])
  personId    Int                   @map("person_id")
  person       Person            @relation(fields: [personId], references: [id])
}
```

this is my query (the important parts):
```TypeScript
const acceptInvite = await prisma.person.update({
    where: {
        id: +req.body.person.id
    },
    data: {
        teamMember: {connect: {id: +req.body.invitation.team.id} }
    },
});
```

Any help would be appreciated. Thanks!

**Comment by kraighamady** [comment]
Okay,  answering my own question for posterity and people like me that apparently need to get off my computer and get some sleep.

That said, for this use case you create on the explicit table directly. In my case above, I would do the following:
```TypeScript
const acceptInvite = await prisma.teamMember.create({
    data: {
        personId: +req.body.person.id,
        teamId: +req.body.invitation.team.id
    },
    include: {
        person: true,
        team: true
    }
});
```
I guess my question now is can this be down on an implicit many to many?  if so, how would you do it?

**Reply by ludralph** [reply]
Hi @kraighamady 

Thank you for your question. If I understand correctly, you would like to accomplish the same use case of connecting a `Team` and a `Person` with an implicit many-to-many relation. This is certainly possible. 

Assuming that your schema looks like this:
```prisma
model Team {
  id      Int      @id @default(autoincrement())
  persons Person[]
}

model Person {
  id    Int    @id @default(autoincrement())
  teams Team[]
}
```

You can create the connection using the code snippet below:
```js
const team = await prisma.team.create({})
const person = await prisma.person.create({})

const personAndTeam = await prisma.person.update({
  where:{
    id: person.id,
  },
  data:{
    teams: {
      connect:{
        id: team.id
      }
    }
  }
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by kraighamady** [comment]
@ludralph I tried that exact method and got an error back saying that it was expecting something (1) and got 0.  I will try to recreate the error so I can update here, but the method above failed for me.

**Reply by ludralph** [reply]
Hi @kraighamady, can you please share a minimal reproduction in a repository so I can take a look?

**Reply by nurul3101** [reply]
@kraighamady Did you had a chance to look at Raphael's response? Please let us know in case you still need any help. If not, please consider marking the discussion as answered/closed.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
